src/elements/content-sidebar/BoxAISidebarContent.tsx

```diff
@@ -0,0 +1,150 @@
+        sendQuestion, 
+        stopQuestion, 
+        ...rest 
+    } = props;
+    const { formatMessage } = useIntl();
+    const { cache, contentName, elementId, recordAction, setCacheValue, userInfo } = React.useContext(BoxAISidebarContext);
+    const { questions: cacheQuestions } = cache;
+
+    if (cache.encodedSession !== encodedSession) {
+        setCacheValue('encodedSession', encodedSession);
+    }
+
+    if (cache.questions !== questions) {
+        setCacheValue('questions', questions);
+    }
+
+    React.useEffect(() => {
+        if (!encodedSession && createSession) {
+            createSession();
+        }
+
+        if (cacheQuestions.length > 0 && cacheQuestions[cacheQuestions.length-1].isCompleted === false) {
+            // if we have cache with question that is not completed resend it to trigger an API
+            sendQuestion({prompt: cacheQuestions[cacheQuestions.length-1].prompt});
+        }
+
+        return () => {
+            // stop API request on unmount (e.g. during switching to another tab)
+            stopQuestion();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const renderBoxAISidebarTitle = () => {
+        return (
+            <div className="bcs-BoxAISidebar-title-part">
+                <Text as="h3" className="bcs-title">
+                    {formatMessage(messages.sidebarBoxAITitle)}
+                </Text>
+                {isAIStudioAgentSelectorEnabled &&
+                    <BoxAiAgentSelectorWithApi
+                        fetcher={getAIStudioAgents}
+                        hostAppName={hostAppName}
+                        onSelectAgent={onSelectAgent}
+                        recordAction={recordAction}
+                        shouldHideAgentSelectorOnLoad
+                        // @ts-ignore variant will be available in higher version
+                        variant="sidebar"
+                    />
+                }
+            </div>
+        );
+    };
+
+    const renderActions = () => (
+        <>
+            { renderBoxAISidebarTitle() }
+            { isResetChatEnabled && 
+                <IconButton
+                    aria-label={formatMessage(sidebarMessages.boxAISidebarClear)}
+                    icon={Trash}
+                    onClick={onClearAction}
+                    size="x-small"
+                />
+            }
+        </>
+    );
+
+    return (
+        <AgentsProvider>
+            <SidebarContent 
+                actions={renderActions()}
+                className="bcs-BoxAISidebar"
+                elementId={elementId}
+                sidebarView={SIDEBAR_VIEW_BOXAI}
+            >
+                <div className="bcs-BoxAISidebar-content">
+                    <BoxAiContentAnswers
+                        className="bcs-BoxAISidebar-contentAnswers"
+                        contentName={contentName}
+                        contentType={formatMessage(messages.sidebarBoxAIContent)}
```

Comment:
shouldnt this be the ext of the file not a translated string?

Reply:
like this prop is used to construct the welcome message based on this contentType which should be an extension so it would says something like "this chat will be cleared when you close this pdf"

Reply:
Yes, however this is what we're currently using in modal in Preview